scripts/prerm

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+rm -f ${CMAKE_INSTALL_FULL_LIBDIR}/python3/dist-packages/TestHarness
+rm -f ${CMAKE_INSTALL_FULL_DATAROOTDIR}/leap_testing/bin
+rmdir --ignore-fail-on-non-empty ${CMAKE_INSTALL_FULL_LIBDIR}/python3/dist-packages
+rmdir --ignore-fail-on-non-empty ${CMAKE_INSTALL_FULL_LIBDIR}/python3
+rmdir --ignore-fail-on-non-empty ${CMAKE_INSTALL_FULL_DATAROOTDIR}/leap_testing
```

Comment:
when/how could this be empty so that it would succeed? Seems impossible: `leap_testing` is part of the package. Feels like we're over complicating this

Reply:
You have a point, for `leap_testing` at least since that is part of the package.  Can probably remove the mkdir from the `postinst` script for `leap_testing` and the `rmdir` for it in `prerm`.

However, I think we need to keep the `python3/dist-packages` as it may be created from scratch in the `postinst` and isn't in the package.  Thus we should also attempt to clean it up.

Reply:
Since the package depends on `python3-numpy` it's impossible for `/usr/lib/python3/dist-packages/` to not exist at the time of `postinst`

Now, of course, we're not linking to `/usr/lib/python3/dist-packages/` but rather `${CMAKE_INSTALL_FULL_LIBDIR}/python3/dist-packages`; that's kind of the disappointment with not being able to use `${Python_SITELIB}` as it would eliminate that academic difference.

Reply:
Simplified per discussion and tested, still works: [e7e66a5](https://github.com/AntelopeIO/leap/pull/818/commits/e7e66a5d1991f60c1f27d4a97e518bfd3a30d9b1)

Reply:
There is no strict guarantee that someone is installing to `/usr/lib/python3/dist-packages` so this solution was formulated to take that into account and still allow it to work.  It also takes into account that with the requirement of running with cmake 3.8, as we discussed, the use of `${Python_SITELIB}` wouldn't currently work as we were intending.

Reply:
There is an important distinction though: while someone may be installing many different versions to various `CMAKE_INSTALL_PREFIX`s via `make dev-install`, they can, by rule, only install a single `leap-dev.deb`. So it seems plausible to consider the installed package's `TestHarness` as _the_ "system provided" `TestHarness` no matter where it is installed.

That actually makes me wonder if instead the `postinst` should be something like
```
ln -s ${CMAKE_INSTALL_FULL_DATAROOTDIR}/leap_testing/tests/TestHarness ${Python_SITELIB}/TestHarness
```
as this will make the installed package's TestHarness the system provided TestHarness no matter what `CMAKE_INSTALL_PREFIX` was used at build time (which should be `/usr` per-spec, but a user might not have configured it that way)

Anyways, we can debate this further post 4.0, where we actually have `${Python_SITELIB}`

Reply:
I was thinking about this a lot, even before you posted this comment.  I tend to agree with you here.  It would be good to treat it that way.  As you said, let's revisit this.  I'll create an issue to track this.

Reply:
@spoonincode - Issue created to investigate this in the future: https://github.com/AntelopeIO/leap/issues/839